### PR TITLE
feat: add stacks with timestamps for the dashboard

### PR DESCRIPTION
**Pull request by Cifko**
Add all stacks into the DB. Add endpoint to the proxy-service for the dashboard to use.
Fill the timestamp with the timestamps from sui, in case of not present, just use local time.

### CODE REVIEW

File: atoma-proxy/src/server/middleware.rs [hunk 1]
```diff
@@ -666,6 +666,15 @@ pub(crate) mod auth {
             .send(AtomaAtomaStateManagerEvent::GetStacksForModel {
                 model: model.to_string(),
                 free_compute_units: total_tokens as i64,
+                owner: sui
```
Comment (jorgeantonio21) [inline atoma-proxy/src/server/middleware.rs:669]: I see, nice !

File: atoma-state/src/migrations/20241213104339_create_initial_schema.sql [hunk 2]
```diff
@@ -45,6 +45,8 @@ CREATE TABLE IF NOT EXISTS stacks (
     in_settle_period BOOLEAN NOT NULL,
     total_hash BYTEA NOT NULL,
     num_total_messages BIGINT NOT NULL,
+    created_at TIMESTAMPTZ NOT NULL,
+    settled_at TIMESTAMPTZ,
```
Comment (jorgeantonio21) [inline atoma-state/src/migrations/20241213104339_create_initial_schema.sql:49]: Not worth creating a new file ?

File: atoma-state/src/state_manager.rs [hunk 3]
```diff
@@ -619,6 +632,38 @@ impl AtomaState {
...
+    /// ```rust,ignore
+    /// use atoma_node::atoma_state::AtomaStateManager;
+    ///
+    /// async fn get_current_stacks(state_manager: &AtomaStateManager) -> Result<Vec<Stack>, AtomaStateManagerError> {
+    ///    state_manager.get_current_stacks().await
+    /// }
+    /// ```
+    #[instrument(level = "trace", skip_all)]
+    pub async fn get_all_stacks(&self) -> Result<Vec<Stack>> {
+        let stacks = sqlx::query("SELECT * FROM stacks")
```
Comment (jorgeantonio21) [inline atoma-state/src/state_manager.rs:658]: There can be a looooot of stacks, can we get them by timestamp and introduce a `limit`parameter ?

**Review by jorgeantonio21 — COMMENTED** [review]
LGTM

Left a comment to address before merge

File: atoma-state/src/state_manager.rs (hunk 3 above)
Comment (Cifko) [reply atoma-state/src/state_manager.rs]: Changed into new table, please take a look

File: atoma-state/src/handlers.rs [hunk 4]
```diff
@@ -381,6 +392,39 @@ pub(crate) async fn handle_stack_created_event(
...
+///
+/// * `Result<()>` - Ok(()) if the event was processed successfully, or an error if something went wrong.
+///
+/// # Errors
+///
+/// This function will return an error if:
+/// * The event data cannot be deserialized into a `StackCreatedEvent`.
+/// * The database operation to insert the new stack fails.
+///
+pub(crate) async fn handle_create_stack_stats(
```
Comment (jorgeantonio21) [inline atoma-state/src/handlers.rs:415]: Add instrumentation

File: atoma-state/src/migrations/20241213104339_create_initial_schema.sql [hunk 5]
```diff
@@ -169,4 +169,14 @@ CREATE TABLE IF NOT EXISTS stats_latency (
     CHECK (date_part('minute', timestamp) = 0 AND date_part('second', timestamp) = 0 AND date_part('milliseconds', timestamp) = 0)
 );
 
-CREATE INDEX IF NOT EXISTS idx_stats_latency ON stats_latency (timestamp);
\ No newline at end of file
+CREATE INDEX IF NOT EXISTS idx_stats_latency ON stats_latency (timestamp);
+
+-- Create stacks table
+CREATE TABLE IF NOT EXISTS stats_stacks (
```
Comment (jorgeantonio21) [inline atoma-state/src/migrations/20241213104339_create_initial_schema.sql:175]: I would add a new file, but ok

File: atoma-state/src/state_manager.rs [hunk 6]
```diff
@@ -3200,6 +3268,25 @@ impl AtomaState {
         .await?;
         Ok(())
     }
+
+    pub async fn new_stats_stack(&self, stack: Stack, timestamp: DateTime<Utc>) -> Result<()> {
```
Comment (jorgeantonio21) [inline atoma-state/src/state_manager.rs:3272]: add docs

**Review by jorgeantonio21 — APPROVED** [review]
LGTM

**Pull request merged**